src/modules/auth/service/auth.service.ts

```diff
@@ -50,17 +55,29 @@ export class AuthService implements IAuthService {
     const hashedPassword = await encryptPassword(password);
     const newUser = { email, password: hashedPassword };
 
-    await this.userService.createUser(newUser);
-
-    // create a token
-    const verificationToken =
-      await this.userEmailVerificationService.createVerificationToken(email);
-
-    // create verification record
-    await this.userEmailVerificationService.createVerification(email, verificationToken);
-
-    // send verification email
-    await this.sendEmailService.sendVerificationEmail(email, verificationToken);
+    const queryRunner = (await this.database.getManager()).connection.createQueryRunner();
+    await queryRunner.startTransaction();
```

Comment:
Thank you for  adding a transaction in here! Should've implemented this earlier 😓